src/app/leverage/components/leverage-widget/index.tsx

```diff
@@ -190,16 +114,22 @@ export function LeverageWidget(props: LeverageWidgetProps) {
       />
       <Summary />
       <Fees costOfCarry={costOfCarry} leverageType={leverageType} />
-      <TradeButton
-        label={buttonLabel}
-        isDarkMode
-        isDisabled={isDisabled}
-        isLoading={isFetchingQuote}
-        onClick={onClickButton}
+      <SmartTradeButton
+        contract={contract ?? ''}
```

Comment:
is there a case where `contract` won't be defined on the `quote` and this is indeed null? if so, maybe we need to add some error handling (I believe empty string will cause an error)